lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -61,9 +61,24 @@ func (d *zfs) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Oper
 		if exists {
 			canRestore := true
 
-			// For block volumes check if the cached image volume is larger than the current pool volume.size
-			// setting (if so we won't be able to resize the snapshot to that the smaller size later).
-			if vol.contentType == ContentTypeBlock {
+			zfsVolType, err := d.getDatasetProperty(d.dataset(vol, true), "type")
+			if err != nil {
+				return err
+			}
+
+			if zfsVolType == "filesystem" && d.isBlockBacking(vol) || zfsVolType == "volume" && !d.isBlockBacking(vol) {
```

Comment:
This stanza could do with a comment explaining why its doing this (like the one below it), as the log message only describes *what* it is doing, as gives little hint to the reader as to the thinking here.

Reply:
Typo "exiting"

Reply:
Typo still here.